skglm/utils.py

```diff
@@ -254,3 +254,55 @@ def check_group_compatible(obj):
                 f"'{obj_name}' is not block-separable. "
                 f"Missing '{attr}' attribute."
             )
+
+
+class AndersonAcceleration:
+    """Abstraction of Anderson Acceleration.
+
+    Extrapolate the asymptotic VAR ``w`` and ``Xw``
+    based on ``K`` previous iterations.
+
+    Parameters
+    ----------
+    K : int
+        Number of previous iterates to consider for extrapolation.
+
+    n_features : int
+        Number of elements in ``w``.
+
+    n_samples: int
+        Number of elements in ``Xw``.
+    """
+
+    def __init__(self, K, n_samples, n_features):
```

Comment:
n_samples and n_features should be inferred and stored at the first call to extrapolate, to make the API lighter